src/frontends/pytorch/src/op/linear.cpp

```diff
@@ -23,21 +23,44 @@ OutputVector translate_linear(const NodeContext& context) {
     num_inputs_check(context, 2, 3);
     auto x = context.get_input(0);
     auto weight = context.get_input(1);
-    if (weight.get_element_type() == element::f16 || weight.get_element_type() == element::bf16) {
+    auto matmul = context.mark_node(std::make_shared<v0::MatMul>(x, weight, false, true));
+    if (!context.input_is_none(2)) {
+        auto bias = context.get_input(2);
+        matmul = context.mark_node(std::make_shared<v1::Add>(matmul, bias));
+    }
+    return {matmul};
+};
+
+OutputVector translate_linear_ext(const NodeContext& context) {
+    num_inputs_check(context, 2, 3);
+    auto x = context.get_input(0);
+    auto initial_x = x;
+    auto weight = context.get_input(1);
+    bool is_compressed = weight.get_element_type() == element::f16 || weight.get_element_type() == element::bf16;
+    bool convert_back = false;
+    if (is_compressed) {
         // In case of patched linear it can have mixed fp16/bf16 and fp32 input type.
         // In other cases these conversion is not required.
-        weight = context.mark_node(std::make_shared<v1::ConvertLike>(weight, x));
+        weight = context.mark_node(std::make_shared<v0::Convert>(weight, element::f32));
+        if (x.get_element_type() != element::f32) {
```

Comment:
can it be that `x` to be of `f64` type?

Reply:
To me, it looks that we need to align types of `x`, `weights` and `bias` to the nearest non-compressed floating-point type

Reply:
We insert this operation in controlled environment and we expect it is always fp32, fp64 is not possible